app/src/main/kotlin/com/simplemobiletools/clock/receivers/DismissAlarmReceiver.kt

```diff
@@ -0,0 +1,51 @@
+package com.simplemobiletools.clock.receivers
+
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import com.simplemobiletools.clock.extensions.*
+import com.simplemobiletools.clock.helpers.ALARM_ID
+import com.simplemobiletools.clock.helpers.NOTIFICATION_ID
+import com.simplemobiletools.clock.models.Alarm
+import com.simplemobiletools.commons.helpers.ensureBackgroundThread
+import java.util.*
+
+class DismissAlarmReceiver: BroadcastReceiver() {
+    override fun onReceive(context: Context, intent: Intent) {
+        val alarmId = intent.getIntExtra(ALARM_ID, -1)
+        val notificationId = intent.getIntExtra(NOTIFICATION_ID, -1)
+        if (alarmId == -1) {
+            return
+        }
+
+        context.hideNotification(notificationId)
+
+        ensureBackgroundThread {
+            context.dbHelper.getAlarmWithId(alarmId)?.let { alarm ->
+                context.cancelAlarmClock(alarm)
+                scheduleNextAlarm(alarm, context)
+                if (alarm.days < 0) {
+                    context.dbHelper.updateAlarmEnabledState(alarm.id, false)
+                    context.updateWidgets()
+                }
+            }
+        }
+
+    }
+
+    private fun scheduleNextAlarm(alarm: Alarm, context: Context) {
+        val oldBitmask = alarm.days
+        alarm.days = removeTodayFromBitmask(oldBitmask)
+        context.scheduleNextAlarm(alarm, false)
+        alarm.days = oldBitmask
+    }
+
+    private fun removeTodayFromBitmask(bitmask: Int): Int {
+        val calendar = Calendar.getInstance()
+        calendar.firstDayOfWeek = Calendar.MONDAY
+        val dayOfWeek = calendar.get(Calendar.DAY_OF_WEEK) - Calendar.MONDAY  // This will give values from 0 (Monday) to 6 (Sunday)
+
+        val todayBitmask = 1 shl dayOfWeek  // This will left shift 0000001 by dayOfWeek places, creating a bitmask for today
+        return bitmask and todayBitmask.inv()  // This will return a new bitmask without today included
```

Comment:
change these 2 lines please, they are way too unreadable

Reply:
What exactly do you want to change here? Do you not want to use the binary operators?, Not using the binary operators would increase the verbosity and make the code less readable in my opinion.
I'm not really sure how to simplify this any further, do you want me to update the comments for the code that I made?

Reply:
lets make it more verbose then, it is always better than not readable code

Reply:
Cool. I've updated the code to not use bit manipulation operators